orchestrator/careplancontributor/taskengine/workflow.go

```diff
@@ -82,6 +85,93 @@ func (f FhirApiWorkflowProvider) QuestionnaireLoader() QuestionnaireLoader {
 	}
 }
 
+var _ WorkflowProvider = &MemoryWorkflowProvider{}
+var _ QuestionnaireLoader = &MemoryWorkflowProvider{}
+
+// MemoryWorkflowProvider is a WorkflowProvider that uses in-memory FHIR resources to provide workflows.
+// To use this provider, you must first load the resources using LoadBundle.
+type MemoryWorkflowProvider struct {
+	questionnaires     []fhir.Questionnaire
+	healthcareServices []fhir.HealthcareService
+}
+
+func (e *MemoryWorkflowProvider) LoadBundle(ctx context.Context, bundleUrl string) error {
+	var bundle fhir.Bundle
+	parsedBundleUrl, err := url.Parse(bundleUrl)
+	if err != nil {
+		return err
+	}
+	client := fhirclient.New(parsedBundleUrl, http.DefaultClient, nil)
+	if err := client.ReadWithContext(ctx, "", &bundle, fhirclient.AtUrl(parsedBundleUrl)); err != nil {
+		return err
+	}
+
+	var questionnaires []fhir.Questionnaire
+	if err := coolfhir.ResourcesInBundle(&bundle, coolfhir.EntryIsOfType("Questionnaire"), &questionnaires); err != nil {
+		return fmt.Errorf("could not extract questionnaires from bundle: %w", err)
+	}
+	e.questionnaires = append(e.questionnaires, questionnaires...)
+
+	var healthcareServices []fhir.HealthcareService
+	if err := coolfhir.ResourcesInBundle(&bundle, coolfhir.EntryIsOfType("HealthcareService"), &healthcareServices); err != nil {
+		return fmt.Errorf("could not extract healthcare services from bundle: %w", err)
+	}
+	e.healthcareServices = append(e.healthcareServices, healthcareServices...)
+	return nil
+}
+
+func (e *MemoryWorkflowProvider) Provide(ctx context.Context, serviceCode fhir.Coding, conditionCode fhir.Coding) (*Workflow, error) {
```

Comment:
With `LoadBundle`, `Provide`, and `Load`, it's not the easiest to determine what does what. Maybe the names could be a bit more descriptive?

Reply:
Provide and Load are from the interfaces, LoadBundle for provisioning the resources required by this type. I'll some godoc.